source/core/MicroBitFiber.cpp

```diff
@@ -412,28 +465,17 @@ int fiber_wait_for_event(uint16_t id, uint16_t value)
   */
 int fiber_wake_on_event(uint16_t id, uint16_t value)
 {
-    Fiber *f = currentFiber;
-
 	if (messageBus == NULL || !fiber_scheduler_running())
 		return MICROBIT_NOT_SUPPORTED;
 
-    // Sleep is a blocking call, so if we're in a fork on block context,
-    // it's time to spawn a new fiber...
-    if (currentFiber->flags & MICROBIT_FIBER_FLAG_FOB)
-    {
-        // Allocate a TCB from the new fiber. This will come from the tread pool if availiable,
-        // else a new one will be allocated on the heap.
-        forkedFiber = getFiberContext();
+    Fiber *f = handle_fob();
 
-        // If we're out of memory, there's nothing we can do.
-        // keep running in the context of the current thread as a best effort.
-        if (forkedFiber != NULL)
-        {
-            f = forkedFiber;
-            dequeue_fiber(f);
-            queue_fiber(f, &runQueue);
-            schedule();
-        }
+    // in case we created a new fiber, make sure to initialize its context
+    // in case schedule() isn't called immedietly afterwards
+    if (f != currentFiber) {
+        dequeue_fiber(f);
+        queue_fiber(f, &runQueue);
+        schedule();
```

Comment:
Hmmm.... I'm not sure this is entirely safe for this particular funciton. fiber_wake_on_event() is used to place the current fiber on a wait queue without entering the scheduler (such that race conditions can be avoided). i.e. 

fiber_wake_on_event(..)
// do something that will cause that event to happen at some later point
schedule();

or am I missing something?

Reply:
oh wait, the schedule() is there already on master; I don't quite understand why, but I think it's fine to align with codal-core